scrapy/core/downloader/handlers/s3.py

```diff
@@ -30,11 +31,11 @@ def _mexe(self, http_request, *args, **kwargs):
     return _S3Connection
 
 
-class S3DownloadHandler(object):
+class S3DownloadHandler:
 
-    def __init__(self, settings, aws_access_key_id=None, aws_secret_access_key=None,
+    def __init__(self, settings, crawler=None,
```

Comment:
Adding a new parameter in between existing ones is strange, I know. But I think adding it between `httpdownloadhandler` and `**kw` might be a bit strange too :man_shrugging: 
Let me know what you think.

Reply:
I think we actually can make a backwards incompatible change here: make all arguments other than settings  keyword-only. In this case order won't matter.

Reply:
Making arguments kwarg-only is the same as this change, but allows to fail earlier.

To make it backwards compatible we can pick crawler from `kw` argument instead.

Reply:
> make all arguments other than settings keyword-only

Nice, I like that. I suppose you're talking about [PEP 3102](https://www.python.org/dev/peps/pep-3102/)? I've seen the syntax but I haven't used it before.